pkg/test/testapplyconfiguration/tester.go

```diff
@@ -162,38 +164,49 @@ func (test *TestOptions) runTest(ctx context.Context) *junitapi.JUnitTestCase {
 		OutputDirectory: test.OutputDirectory,
 		Now:             test.Description.Now.Time,
 	}
-	actualResult, err := applyconfiguration.ExecApplyConfiguration(ctx, test.Description.BinaryName, args)
+	actualResult, execErr := applyconfiguration.ExecApplyConfiguration(ctx, test.Description.BinaryName, args)
 	endTime := now()
 	currJunit.Duration = endTime.Sub(startTime).Round(1 * time.Second).Seconds()
 
 	switch {
-	case err == nil && actualResult != nil:
+	case execErr == nil && actualResult != nil:
 		// this was successful
-	case err == nil && actualResult == nil:
+	case execErr == nil && actualResult == nil:
 		currJunit.FailureOutput = &junitapi.FailureOutput{
 			Message: "No result or error from apply-configuration",
 			Output:  "No result or error from apply-configuration",
 		}
 		return currJunit
 
-	case err != nil && actualResult != nil:
+	case execErr != nil && actualResult != nil:
 		currJunit.SystemOut = actualResult.Stdout()
 		currJunit.SystemErr = actualResult.Stderr()
 		fallthrough
-	case err != nil && actualResult == nil:
-		currJunit.FailureOutput = &junitapi.FailureOutput{
-			Message: fmt.Sprintf("%v\n%v", err, currJunit.SystemErr),
-			Output:  fmt.Sprintf("ERROR:%v\n\nSTDERR:\n%s\n\nSTDOUT:\n:%s\n", err, currJunit.SystemErr, currJunit.SystemOut),
+	case execErr != nil && actualResult == nil:
+		switch {
+		case test.Description.DesiredError == NonZeroReturn:
+			var exitErr *exec.ExitError
+			if !errors.As(execErr, &exitErr) {
+				// TODO figure if more verification makes sense.
+				// TODO probably add inspection of stderr and stdout for comparison
+				exitErr.ExitCode()
```

Comment:
this will blow up, unless you change the condition.

Reply:
> this will blow up, unless you change the condition.

?  This is just a function that returns the exit code of the exec'd process

Reply:
ah, casting.